test/components/table/list-view.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+import test from 'ava';
+import Adapter from 'enzyme-adapter-react-16';
+import React from 'react';
+import {shallow, configure} from 'enzyme';
+import {listGranules} from '../../../app/scripts/actions';
+import List from '../../../app/scripts/components/table/list-view';
+import Timer from '../../../app/scripts/components/app/timer.js';
+import {
+  errorTableHeader,
+  errorTableRow, errorTableSortProps
+} from '../../../app/scripts/utils/table-config/granules';
+
+configure({ adapter: new Adapter() });
+
+test('list-view should properly initialize timer config prop', async (t) => {
+  const dispatch = () => Promise.resolve();
+  const query = { q: '_exists_:error AND status:failed' };
+  const list = {
+    meta: {},
+    data: []
+  };
+  const listWrapper = shallow(
+    <List
+      list={list}
+      dispatch={dispatch}
+      action={listGranules}
+      tableHeader={errorTableHeader}
+      sortIdx={4}
+      tableRow={errorTableRow}
+      tableSortProps={errorTableSortProps}
+      query={query}
+    />,
+    {
+      // We must disable lifecycle methods so that when we find the Timer its
+      // properties will still be set to their initial values.  Otherwise, a
+      // lifecycle method might cause the properties to change and we will not
+      // be able to check their initial values, which is what we want to do.
+      disableLifecycleMethods: true,
+      context: {
+        store: {
```

Comment:
So is it only working with you when you provide the store this way instead of via the `Provider`? If so, I don't care or want it to change, it's just weird

Reply:
Yeah, I don't know why using the Provider with a store didn't work. In that case, the Timer couldn't be found. I could have added the store as a property on the List component, but chose to add it to the options context instead so that the List component appears just like it does in our usage of it.